test/with_dummyserver/test_poolmanager.py

```diff
@@ -378,6 +379,31 @@ def test_top_level_request_with_body(self):
         assert r.status == 200
         assert r.data == b"test"
 
+    @pytest.mark.parametrize(
+        "headers", [None, {"content-Type": "application/json"}, {"attribute": "value"}]
```

Comment:
Would be good to have tests for different casings of "Content-Length" and with an HTTPHeaderDict too.

We should also make some assertation that the content-length header is sent and if already set to a value won't be overwritten.

Reply:
Sure